clients/fides-js/src/lib/i18n/i18n-utils.ts

```diff
@@ -77,6 +78,66 @@ function extractMessagesFromExperienceConfig(
   return extracted;
 }
 
+/**
+ * Helper function to extract all the translated messages from the "gvl_translations"
+ * API response. Returns an object that maps locales -> messages, e.g.
+ * {
+ *   "en": {
+ *     "exp.tcf.purposes.1.name": "Store and/or access information on a device",
+ *     "exp.tcf.purposes.1.description": "description": "Cookies, device or similar...",
+ *     "exp.tcf.purposes.1.illustrations.0": "Most purposes explained in this...",
+ *   },
+ *   "es": {
+ *     ...
+ *   }
+ * }
+ *
+ * This allows convenient access to any translated message from all points in
+ * the TCF code by using the well-worn GVL IDs as locators. However, note that
+ * the "illustrations" array can be a bit awkward to use as a flattened list of
+ * messages - we need to make sure we carefully look up illustrations using
+ * their zero-based index (e.g. i18n.t("exp.tcf.purposes.1.illustrations.0")
+ */
+function extractMessagesFromGVLTranslations(
```

Comment:
do we have a test to explicitly test GVL --> messages or did I miss it?

Reply:
Yup! In two places actually:
1. Unit tests for the "extract" logic populating the i18n message catalog: https://github.com/ethyca/fides/pull/4677/files#diff-7f4b8f4695faf7ecff210470bda8a5da59ce294818d9706b9b50ae1de8875441R200
2. E2E tests to see that these GVL strings are properly displayed; here's an example string that comes exclusively from GVL translations for example: https://github.com/ethyca/fides/pull/4677/files#diff-d2fd7f7c709639c5342597c928b1e002136a1c0936f558d3c144f2781a0c6af2R297